internal/provider/datasources/data_source_api_tokens_test.go

```diff
@@ -0,0 +1,175 @@
+package datasources_test
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients/iam"
+
+	astronomerprovider "github.com/astronomer/terraform-provider-astro/internal/provider"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+)
+
+func TestAcc_DataSourceApiTokens(t *testing.T) {
+	tfVarName := "test_data_api_tokens"
+	tfOrganizationId := os.Getenv("HOSTED_ORGANIZATION_ID")
+	tfWorkspaceId := os.Getenv("HOSTED_WORKSPACE_ID")
+	tfDeploymentId := os.Getenv("HOSTED_DEPLOYMENT_ID")
+	tfOrgOnly := true
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			astronomerprovider.TestAccPreCheck(t)
+		},
+		ProtoV6ProviderFactories: astronomerprovider.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiTokens(tfVarName),
+				Check: resource.ComposeTestCheckFunc(
+					checkApiTokens(tfVarName, false, "", false, "", false, tfOrganizationId),
+				),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiTokensFilterWorkspaceId(tfVarName, tfWorkspaceId),
+				Check: resource.ComposeTestCheckFunc(
+					checkApiTokens(tfVarName, true, tfWorkspaceId, false, "", false, tfOrganizationId),
+				),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiTokensFilterDeploymentId(tfVarName, tfDeploymentId),
+				Check: resource.ComposeTestCheckFunc(
+					checkApiTokens(tfVarName, false, "", true, tfDeploymentId, false, tfOrganizationId),
+				),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + apiTokensFilterOrgOnly(tfVarName, tfOrgOnly),
+				Check: resource.ComposeTestCheckFunc(
+					checkApiTokens(tfVarName, false, "", false, "", true, tfOrganizationId),
+				),
+			},
+		},
+	})
+}
+
+func apiTokens(tfVarName string) string {
+	return fmt.Sprintf(`
+data astro_api_tokens "%v" {}`, tfVarName)
+}
+
+func apiTokensFilterWorkspaceId(tfVarName string, workspaceId string) string {
+	return fmt.Sprintf(`
+data astro_api_tokens "%v" {
+	workspace_id = "%v"
+}`, tfVarName, workspaceId)
+}
+
+func apiTokensFilterDeploymentId(tfVarName string, deploymentId string) string {
+	return fmt.Sprintf(`
+data astro_api_tokens "%v" {
+	deployment_id = "%v"
+}`, tfVarName, deploymentId)
+}
+
+func apiTokensFilterOrgOnly(tfVarName string, orgOnly bool) string {
+	return fmt.Sprintf(`
+data astro_api_tokens "%v" {
+	include_only_organization_tokens = %v
+}`, tfVarName, orgOnly)
+}
+
+func checkApiTokens(tfVarName string, filterWorkspaceId bool, workspaceId string, filterDeploymentId bool, deploymentId string, filterOrgOnly bool, organizationId string) resource.TestCheckFunc {
```

Comment:
lets make an input struct for this function
its getting a lil long with the number of params